java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogStateStore.java

```diff
@@ -36,11 +39,35 @@ public DynamoDBTransactionLogStateStore(
 
     public static TransactionLogStateStore.Builder builderFrom(
             InstanceProperties instanceProperties, TableProperties tableProperties, AmazonDynamoDB dynamoDB, AmazonS3 s3) {
-        return builder()
+        return builderFrom(instanceProperties, tableProperties, dynamoDB, s3, new Configuration());
+    }
+
+    public static TransactionLogStateStore.Builder builderFrom(
+            InstanceProperties instanceProperties, TableProperties tableProperties, AmazonDynamoDB dynamoDB, AmazonS3 s3, Configuration configuration) {
+        Builder builder = builder()
                 .sleeperTable(tableProperties.getStatus())
                 .schema(tableProperties.getSchema())
                 .filesLogStore(new DynamoDBTransactionLogStore(instanceProperties.get(FILE_TRANSACTION_LOG_TABLENAME), instanceProperties, tableProperties, dynamoDB, s3))
                 .partitionsLogStore(new DynamoDBTransactionLogStore(instanceProperties.get(PARTITION_TRANSACTION_LOG_TABLENAME), instanceProperties, tableProperties, dynamoDB, s3));
+        if (instanceProperties.getBoolean(TRANSACTION_LOG_STATE_STORE_LOAD_LATEST_SNAPSHOTS)) {
+            loadLatestSnapshots(builder, instanceProperties, tableProperties, dynamoDB, configuration);
+        }
+        return builder;
     }
 
+    private static void loadLatestSnapshots(
+            TransactionLogStateStore.Builder builder, InstanceProperties instanceProperties, TableProperties tableProperties,
+            AmazonDynamoDB dynamoDB, Configuration configuration) {
+        new DynamoDBTransactionLogSnapshotStore(instanceProperties, tableProperties, dynamoDB).getLatestSnapshots()
+                .ifPresent(latestSnapshots -> {
+                    TransactionLogFilesSnapshotSerDe filesSnapshotSerDe = new TransactionLogFilesSnapshotSerDe(configuration);
+                    TransactionLogPartitionsSnapshotSerDe partitionsSnapshotSerDe = new TransactionLogPartitionsSnapshotSerDe(tableProperties.getSchema(), configuration);
+                    try {
+                        builder.filesState(filesSnapshotSerDe.load(latestSnapshots.getFilesSnapshot()))
+                                .partitionsState(partitionsSnapshotSerDe.load(latestSnapshots.getPartitionsSnapshot()));
```

Comment:
This won't skip to the right place in the transaction log. Please add some tests to ensure the transaction logs are not read before the transaction numbers held against the snapshots.